### ISSUE TITLE
Prerendering with +server.js blocks cURL requests

### ISSUE THREAD

**Issue by dextermb** [sveltejs/kit#10102]
### Describe the bug

When attempting to cURL a preview build of SvelteKit that uses `prerender` within a `+server.js` file a `405 Method not allowed` is returned. However when providing an extra "Accept" header `200 OK` is returned, and when using `+page.server.js` instead a `200 OK` is returned.

Example repository linked below.

### Reproduction

https://github.com/dextermb/svelte-kit-prerender-405-server-js

### System Info

```Shell
npx envinfo --system --binaries --browsers --npmPackages "{svelte,@sveltejs/*,vite}"
Need to install the following packages:
  envinfo@7.8.1
Ok to proceed? (y) y

  System:
    OS: macOS 13.4
    CPU: (12) arm64 Apple M2 Max
    Memory: 79.95 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
  Browsers:
    Chrome: 113.0.5672.126
    Firefox: 110.0.1
    Safari: 16.5
```

### Severity

annoyance

**Comment by tcc-sejohnson** [comment]
Ahhh, okay, yeah -- I don't believe this is a bug. There are two things at play here:

1. You can't prerender `+server` files with mutative methods (`POST`, for example) -- this doesn't apply here, but does apply to the above PR
    - @gtm-nayan In light of this, I'm not sure if your PR is necessary -- it certainly shouldn't be for prerendering, though if it fixes some undiscovered bug for runtime code, we might still want to merge it
2. You can't prerender a route with both a `+server` file and a `+page` file. This is because content negotiation is impossible for prerendered files -- there's no server logic to disambiguate a request to `/foo` using the `Accept` header
    - I think you're seeing mixed success here right now because the preview server is still performing this content negotiation for you when it shouldn't be?

This PR will throw an error when you try to do this: https://github.com/sveltejs/kit/pull/9994/files

**Comment by dummdidumm** [comment]
Closed via #11256, having conflicting route definitions / both a `+page.js` and `+server.js` when prerendering is an error in SvelteKit 2.0
